### PR TITLE
Hott755 dockerise backend

### PR DESCRIPTION
**Pull request by mafonso**
### Jira link

HOTT-755

### What?

I have added/removed/altered:

- [ ] Build docker image for the backend app (DEV only)
- [ ] Deploy docker backend service in blue-green (DEV only)
- [ ] Deploy docker based worker (DEV only)

### Why?

I am doing this because:
- Moving to docker based deployment 

Note before merging:  This PR is using temporary target apps (-docker suffixed) to avoid clash with the current ones. 
This before merging, the apps need to be renamed in CF and this PR edited accordingly.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Build compilation image
+FROM ruby:2.7.4-alpine3.13 as builder
```
Comment (willfish) [inline Dockerfile:2]:
```suggestion
FROM ruby:3.0.2-alpine3.13 as builder
```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+RUN rm -rf node_modules log tmp && \
+      rm -rf /usr/local/bundle/cache && \
+      rm -rf .env && \
+      find /usr/local/bundle/gems -name "*.c" -delete && \
+      find /usr/local/bundle/gems -name "*.h" -delete && \
+      find /usr/local/bundle/gems -name "*.o" -delete && \
+      find /usr/local/bundle/gems -name "*.html" -delete
+
+# Build runtime image
+FROM ruby:2.7.4-alpine3.13 as production
```
Comment (willfish) [inline Dockerfile:37]:
```suggestion
FROM ruby:3.0.2-alpine3.13 as production
```

Comment (willfish) [reply Dockerfile:37]: Does the latest alpine.x.xx image work these days? Might be worth bumping?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+# Copy all files to /app (except what is defined in .dockerignore)
+COPY . /app/
+
+
+# Cleanup to save space in the production image
+RUN rm -rf node_modules log tmp && \
+      rm -rf /usr/local/bundle/cache && \
+      rm -rf .env && \
+      find /usr/local/bundle/gems -name "*.c" -delete && \
```
Comment (willfish) [inline Dockerfile:31]: Can you achieve this in one with a regex?

Comment (mafonso) [reply Dockerfile:31]: Probably, but I think it clearer this way, as each line is targeting a specif "left over". 
Might not be worth the time getting it right, and there will be some risk.

**Comment by mafonso** [comment]
We bumped the docker version in Dockerfile, but the Gemfile is still behind. 
Maybe it's best to deal with the Ruby version upgrade in a separate PR?

**Review by willfish — APPROVED** [review]
LGTM

**Pull request merged**